astropy/io/fits/src/compressionmodule.c

```diff
@@ -233,159 +235,136 @@ int compress_type_from_string(char* zcmptype) {
 }
 
 
+PyObject *
+get_header_value(PyObject* header, char* key) {
+    PyObject* hdrkey;
+    PyObject* hdrval;
+    hdrkey = PyUnicode_FromString(key);
+    if (hdrkey == NULL) {
+        return NULL;
+    }
+    hdrval = PyObject_GetItem(header, hdrkey);
+    Py_DECREF(hdrkey);
+    PyErr_Clear();
+    return hdrval;
+}
+
+
 // TODO: It might be possible to simplify these further by making the
 // conversion function (eg. PyString_AsString) an argument to a macro or
 // something, but I'm not sure yet how easy it is to generalize the error
 // handling
+/* The get_header_* functions resemble "Header.get" where "def" is the default
+   value, "keyword" is a string representing the header-key and the result is
+   stored in "val".
+   The function returns 0 on success, 1 if the header didn't have the keyword
+   and the default was applied and -1 (with an exception set) if an Exception
+   happened (like a MemoryError or Overflow).
+*/
 int get_header_string(PyObject* header, char* keyword, char* val, char* def) {
-    PyObject* keystr;
-    PyObject* keyval;
-    PyObject* tmp;  // Temp pointer to decoded bytes object
-    int retval;
-
-    keystr = PyUnicode_FromString(keyword);
-    keyval = PyObject_GetItem(header, keystr);
+    PyObject* keyval = get_header_value(header, keyword);
 
     if (keyval != NULL) {
+        PyObject* tmp = PyUnicode_AsLatin1String(keyval);
         // FITS header values should always be ASCII, but Latin1 is on the
         // safe side
-        tmp = PyUnicode_AsLatin1String(keyval);
+        Py_DECREF(keyval);
+        if (tmp == NULL) {
+            /* could always fail to allocate the memory or such like. */
+            return -1;
+        }
         strncpy(val, PyBytes_AsString(tmp), 72);
```

Comment:
It might be nice to "un-magic" the number `72` with a `#define`.

Reply:
okay, I was too fast with the update. Could you repost the comment?